sonarts-core/src/tools/newRule.ts

```diff
@@ -1,269 +1,286 @@
+    const linksToTitles: { [x: string]: string } = {};
+    for (let i = 0; i < ruleTitles.length; i++) {
+      linksToTitles[ruleLinks[i]] = ruleTitles[i];
+    }
 
-    let ruleData = JSON.parse(fileText);
-    return { ruleTitle: ruleData["title"], rspecKey: ruleData["ruleSpecification"] };
-  } catch (err) {
-    console.error("could not find metadata for the rule. Have you run the rspec-api (with correct parameters)?");
-    showHelp();
-    throw err;
-  }
-}
+    const sortedRuleLinks = Object.keys(linksToTitles).sort();
+    const sortedRuleTitles = sortedRuleLinks.map(ruleKey => linksToTitles[ruleKey]);
 
-function getDescription(rspecRuleFolder: string, rspecId: string) {
-  try {
-    let replaceDict: any = {};
-    replaceDict["<em>"] = "";
-    replaceDict["</em>"] = "";
-    replaceDict["<p>"] = "";
-    replaceDict["</p>"] = `\n`;
-    replaceDict["<code>"] = "`";
-    replaceDict["</code>"] = "`";
-    replaceDict["<pre>"] = "```typescript";
-    replaceDict["</pre>"] = "```";
-    replaceDict["<h2>"] = "## ";
-    replaceDict["</h2>"] = `\n`;
-
-    const fileText = fs.readFileSync(`${rspecRuleFolder}/${rspecId}.html`).toString();
-    return replace(fileText, replaceDict);
-  } catch (err) {
-    console.error("could not find metadata for the rule. Have you run the rspec-api?");
-    showHelp();
-    throw err;
+    fs.writeFileSync(readmePath, [...head, ...sortedRuleTitles, "", ...sortedRuleLinks, "", ...tail].join("\n"));
   }
-}
 
-function updateReadme(rootFolder: string, ruleTitle: string, ruleNameDash: string) {
-  const readmePath = `${rootFolder}/README.md`;
-  let readme = fs.readFileSync(readmePath).toString();
-
-  const lines = readme.split("\n");
-
-  let result = [];
-
-  let resultsBlock1 = [];
-  let resultsBlock2 = [];
-  let resultsBlock3 = [];
-
-  let ruleValue = `* ${ruleTitle} ([\`${ruleNameDash}\`])`;
-  resultsBlock1.push(ruleValue);
-  let ruleKey = `[\`${ruleNameDash}\`]: ./sonarts-core/docs/rules/${ruleNameDash}.md`;
-  resultsBlock2.push(ruleKey);
-
-  // 0: start
-  // 1: ## Rules
-  // 2: newlines
-  // 3: rules
-  // 4: newlines
-  // 5: rules 2
-  // 6: newline
-  // 7: after
-
-  let state = 0;
-
-  for (const line of lines) {
-    switch (state) {
-      case 0:
-        if (line.trim() === "## Rules") {
-          state = 1;
-        }
-        result.push(line);
-        break;
-
-      case 1:
-      case 2:
-        if (line.length > 0) {
-          state++;
-          resultsBlock1.push(line);
-        } else {
-          result.push(line);
-        }
-
-        break;
-
-      case 3:
-        if (line.length === 0) {
-          state++;
-        } else {
-          resultsBlock1.push(line);
-        }
-        break;
-
-      case 4:
-        if (line.length === 0) {
-          state++;
-        } else {
-          resultsBlock2.push(line);
-        }
-        break;
-      case 5:
-        resultsBlock3.push(line);
-        break;
+  function parseReadme() {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: The function parseReadme has a cyclomatic complexity of 11 which is higher than the threshold of 10 ([typescript:S1541](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS1541))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=sonarts&branch=stas%2Frefactor-new-rule&issues=AWDF-9c2_Q-0dz4JA6cy&open=AWDF-9c2_Q-0dz4JA6cy)

[ISSUE_KEY]: AWDF-9c2_Q-0dz4JA6cy